beetsplug/spotify.py

```diff
@@ -154,27 +158,32 @@ def _authenticate(self):
         self._log.debug(
             "{} access token: {}", self.data_source, self.access_token
         )
-        with open(self.tokenfile, "w") as f:
+        with open(self._tokenfile(), "w") as f:
             json.dump({"access_token": self.access_token}, f)
 
     def _handle_response(
-        self, request_type, url, params=None, retry_count=0, max_retries=3
-    ):
+        self,
+        method: Literal["get", "post", "put", "delete"],
+        url: str,
+        params: Any = None,
+        retry_count=0,
+        max_retries=3,
```

Comment:
While you're at it, maybe also add the remaining argument types?

```suggestion
        retry_count: int = 0,
        max_retries: int = 3,
```